File: WebApi/Database/Entities/Flow/Asset.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace Database.Entities
{
    public class Asset : ValueObject
    {
        public decimal Interest { get; set; } // Odsetki roczne
        public Rate InterestRate { get; set; } = Rate.Yearly;
        public AssetType AssetType { get; set; } = AssetType.Undefined;
        public string? Ticker { get; set; } // Stock ticker symbol (e.g., "EOSE" for EOS Energy)

        [NotMapped]
        public bool HasTicker => AssetType == AssetType.Stocks && !string.IsNullOrWhiteSpace(this.Ticker);
    }
}

File: Website/src/app/models/asset.ts
import { Rate } from './rate';
import { ValueObject } from './valueObject';
import { AssetType } from './assetType';

export interface Asset extends ValueObject {
  interest: number;
  interestRate: Rate;
  assetType: AssetType;
  ticker?: string; // Stock ticker symbol (e.g., "EOSE" for EOS Energy)

  // View-model
  isEditing: boolean;
}
